app/models/visitors_validation.rb

```diff
@@ -0,0 +1,47 @@
+class VisitorsValidation
+  include NonPersistedModel
+
+  MIN_LEAD_VISITOR_AGE = 18 # years
+
+  attribute :lead_date_of_birth, Date
+  attribute :dates_of_birth, Array[Date]
+  attribute :prison, Prison
+
+  validate :lead_visitor_age
+  validate :number_of_visitors
+  validate :number_of_adults
+
+  def lead_visitor_age
+    if age_calculator.age(lead_date_of_birth) < MIN_LEAD_VISITOR_AGE
+      errors.add(:general, 'lead_visitor_age')
+    end
+  end
+
+  def number_of_visitors
+    if dates_of_birth.size > Prison::MAX_VISITORS
+      errors.add(:general, 'too_many_visitors')
```

Comment:
Using strings matters a lot. When using symbols the error message is localised when fetching the values on `errors`, had me confused for a while.